src/taipy/core/config/data_node_config.py

```diff
@@ -267,15 +296,26 @@ def _update(self, as_dict, default_section=None):
         self._storage_type = as_dict.pop(self._STORAGE_TYPE_KEY, self._storage_type)
         if self._storage_type is None and default_section:
             self._storage_type = default_section.storage_type
+
         self._scope = as_dict.pop(self._SCOPE_KEY, self._scope)
         if self._scope is None and default_section:
-            self._scope = default_section.scope
+            if default_section.scope:
+                self._scope = default_section.scope
+            else:
+                self._scope = self._DEFAULT_SCOPE
+
         self._properties.update(as_dict)
-        if default_section:
+        if default_section and self._storage_type == default_section.storage_type:
             self._properties = {**default_section.properties, **self._properties}
 
+        # Assign default value to optional properties if not defined by user
```

Comment:
This section is added because when configure a datanode, we ignore all optional fields that have `None` value.

Then when compile the Config, we check if that optional property is defined by user or not to assign to the data node.
If not, then use the pre-defined value on `_OPTIONAL_PROPERTIES`.